src/test/java/com/uid2/admin/vertx/OperatorKeyServiceTest.java

```diff
@@ -0,0 +1,98 @@
+package com.uid2.admin.vertx;
+
+import com.uid2.admin.model.Site;
+import com.uid2.admin.vertx.service.IService;
+import com.uid2.admin.vertx.service.OperatorKeyService;
+import com.uid2.admin.vertx.test.ServiceTestBase;
+import com.uid2.shared.auth.OperatorKey;
+import com.uid2.shared.auth.Role;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.verify;
+
+public class OperatorKeyServiceTest extends ServiceTestBase {
+    @Override
+    protected IService createService() {
+        return new OperatorKeyService(config, auth, writeLock, storageManager, operatorKeyProvider, siteProvider, keyGenerator);
+    }
+
+    private void checkOperatorKeyResponse(OperatorKey[] expectedOperators, Object[] actualOperators) {
+        assertEquals(expectedOperators.length, actualOperators.length);
+        for (int i = 0; i < expectedOperators.length; ++i) {
+            OperatorKey expectedOperator = expectedOperators[i];
+            JsonObject actualOperator = (JsonObject) actualOperators[i];
+            assertEquals(expectedOperator.getName(), actualOperator.getString("name"));
+            assertEquals(expectedOperator.getContact(), actualOperator.getString("contact"));
+            assertEquals(expectedOperator.getProtocol(), actualOperator.getString("protocol"));
+            assertEquals(expectedOperator.isDisabled(), actualOperator.getBoolean("disabled"));
+            assertEquals(expectedOperator.getSiteId(), actualOperator.getInteger("site_id"));
+        }
+    }
+
+    @Test
+    void operatorAdd(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.OPERATOR_MANAGER);
+        setSites(new Site(5, "test_site", true));
+        OperatorKey[] expectedOperators = {
+                new OperatorKey("", "test_operator", "test_operator", "trusted", 0, false, 5)
+        };
+
+        post(vertx, "api/operator/add?name=test_operator&protocol=trusted&site_id=5", "", ar -> {
+            assertTrue(ar.succeeded());
+            HttpResponse response = ar.result();
+            assertEquals(200, response.statusCode());
+            checkOperatorKeyResponse(expectedOperators, new Object[]{response.bodyAsJsonObject()});
+
+            try {
+                verify(storageManager).uploadOperatorKeys(any(), collectionOfSize(1));
+            } catch (Exception ex) {
+                fail(ex);
+            }
+
+            testContext.completeNow();
+        });
+    }
+
+    @Test
+    void clientAddUnknownSiteId(Vertx vertx, VertxTestContext testContext) {
```

Comment:
these are still tests to test adding/changing operator/operatorKey, the term "client" is a bit confusing as it seems to refer to Client Key so maybe rename it back to operatorAddUnknownSiteId instead?

Reply:
Sure, will updating this